Fixed build for clang16 and lto

Schievel1 opened the pull request:
Clang16 will not allow implicit int, implicit function declaration etc. by default.
This overhauls the whole source code to work with clang16 default.

Also fixes some -Werror=lto-type-mismatch

I also used modern C function definitions everywhere.

See also:
https://bugs.gentoo.org/870907
https://bugs.gentoo.org/861467

Signed-off-by: Pascal Jäger <pascal.jaeger@leimstift.de>

johnsonjh commented:
Nice job - I'll review this a bit more later and see if I have any feedback, but, at first glance, I see there are a few .orig and .rej files that crept in.  Can you remove those?

Schievel1 commented:
Yes I noticed, that’s why I reverted it to being a draft again. I will remove them, don’t worry. 

Please review thoroughly, I have tested this and I can run scrollz, I also connected to a server and wrote messages. But I haven’t tested every single function.

Also take your time, release of clang16 is 2024 I think

johnsonjh commented:
No problem.  It's a big task!

Just a hint, use `SZdist` and test all the various option combinations as well, just be sure.

Of course, Flier will have the final word on it, but if it doesn't change or break any functionality it should pass his review, if all done well.

On Tue, Nov 15, 2022, at 1:38 PM, Pascal J wrote:
> 
> 
> Yes I noticed, that’s why I reverted it to being a draft again. I will remove them, don’t worry.
> 
> Please review thoroughly, I have tested this and I can run scrollz, I also connected to a server and wrote messages. But I haven’t tested every single function.
> 
> Also take your time, release of clang16 is 2024 I think
> 
> 
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/ScrollZ/ScrollZ/pull/30#issuecomment-1315717271>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AOWGFJQX2BIOFLS3VFKKGCDWIPKB3ANCNFSM6AAAAAASBD6FDU>.
> You are receiving this because you commented.Message ID: ***@***.***>
> 
>

Schievel1 committed:
Fix function declarations for clang16 and LTO

Clang16 will not allow implicit int, implicit function declaration etc. by default.
This overhauls the whole source code to work with clang16 default.

Also fixes some -Werror=lto-type-mismatch

I also used modern C function definitions everywhere.

See also:
https://bugs.gentoo.org/870907
https://bugs.gentoo.org/861467

Signed-off-by: Pascal Jäger <pascal.jaeger@leimstift.de>

Schievel1 commented:
Oh and I also wrote a mail to flier@scrollz.info because I did not know there is a repo on github. So, please ignore the mail...

johnsonjh commented:
Bump (for my TODO list)

johnsonjh commented:
Re-bumping. :)

justflier commented:
Applied manually, thanks!
